src/main/java/com/amuzil/omegasource/magus/skill/modifiers/data/HeldModifierData.java

```diff
@@ -0,0 +1,55 @@
+package com.amuzil.omegasource.magus.skill.modifiers.data;
+
+import com.amuzil.omegasource.magus.skill.modifiers.api.ModifierData;
+import net.minecraft.nbt.CompoundTag;
+import org.apache.logging.log4j.LogManager;
+
+public class HeldModifierData extends BaseModifierData {
```

Comment:
I'm not sure about having to pair each Listener with a Data class...
Actually it makes sense; Condition style Modifier, with a Data trait to store it